### PR TITLE
[12.0][MIG] stock_account_internal_move

### PR DESCRIPTION
**Pull request by kittiu**
Migrate to v12

### CODE REVIEW

**Comment by kittiu** [comment]
Migrated

**Review by rousseldenis — APPROVED** [review]
Code review

File: stock_account_internal_move/models/stock_move.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        return res
+
+    # @override
+    @api.multi
+    def _run_valuation(self, quantity=None):
+        # Extend `_run_valuation` to make it work on internal moves.
+        self.ensure_one()
+        res = super()._run_valuation(quantity)
+        if self._is_internal() and not self.value:
+            # TODO: recheck if this part respects product valuation method
```
Comment (tbaden) [inline stock_account_internal_move/models/stock_move.py:34]: were the todo checked?

**Review by tbaden — COMMENTED** [review]
thanks for the work 👍 
code review: lgtm 
just the one question (see below)

File: stock_account_internal_move/models/stock_move.py (hunk 1 above)
Comment (kittiu) [reply stock_account_internal_move/models/stock_move.py:34]: Hi, sorry, I am not sure what is the intention of the original author too.
Any hint, @max3903?

**Review by gurneyalex — APPROVED** [review]

File: stock_account_internal_move/models/stock_location.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @api.constrains(
+        'usage',
+        'force_accounting_entries',
+    )
+    def _check_force_accounting_entries_internal_only(self):
+        for loc in self:
+            if loc.usage != 'internal' and loc.force_accounting_entries:
+                raise ValidationError(_(
+                    'You cannot force accounting entries'
+                    ' on a non-internal locations.'))
```
Comment (gurneyalex) [inline stock_account_internal_move/models/stock_location.py]:
```suggestion
                    ' on a non-internal location.'))
```

File: stock_account_internal_move/models/stock_location.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    )
+    def _check_internal_valuation_accounts_present(self):
+        """Ensure that every location requiring entries has valuation accs."""
+        for loc in self:
+            if loc.usage != 'internal' or not loc.force_accounting_entries:
+                continue  # this one doesn't require accounts, it's fine
+            if not loc.valuation_in_account_id \
+                    or not loc.valuation_out_account_id:
+                raise ValidationError(_(
+                    'You must provide a valuation in/out accounts'
```
Comment (gurneyalex) [inline stock_account_internal_move/models/stock_location.py]:
```suggestion
                    'You must provide a valuation in/out account'
```

**Review by gurneyalex — COMMENTED** [review]
also suggested some typo fixes in the strings

**Comment by rousseldenis** [comment]
@kittiu Do you plan to attend @gurneyalex comments ?

**Comment by kittiu** [comment]
> @kittiu Do you plan to attend @gurneyalex comments ?

Done :)

**Comment by rousseldenis** [comment]
@max3903 Any advice on the TODO ?

**Comment by rousseldenis** [comment]
I think we can merge this.

**Pull request merged**